Formula/dtools.rb

```diff
@@ -0,0 +1,73 @@
+class Dtools < Formula
+  desc "D programming language tools"
+  homepage "https://dlang.org/"
+  url "https://github.com/dlang/tools/archive/refs/tags/v2.103.0.tar.gz"
+  sha256 "591bf56d7c8aa45205a3533438fef5bd48007756446f5cf032fcabcc077afdd1"
+  license "BSL-1.0"
+  head "https://github.com/dlang/tools.git", branch: "master"
+
+  depends_on "dub" => :build
+  depends_on "ldc" => [:build, :test]
+
+  link_overwrite "bin/ddemangle"
+  link_overwrite "bin/dustmite"
+  link_overwrite "bin/rdmd"
+
+  def install
+    # We only need the "public" tools, as listed at
+    # https://github.com/dlang/tools/blob/master/README.md
+    #
+    # Skip building dman as it requires getting and building the DMD
+    # and dlang.org source trees.
+    tools = %w[ddemangle rdmd]
+    system "dub", "add-local", buildpath
+
+    tools.each do |tool|
+      system "dub", "build", "--build=release", ":#{tool}"
+      bin.install "dtools_#{tool}" => tool
+    end
+
+    # DustMite is not provided as a dub target
+    system "ldc2", "-O", "--release", *Dir.glob("DustMite/*.d"), "-od=build", "-of=#{bin}/dustmite"
+
+    man1.install "man/man1/rdmd.1"
+  end
+
+  test do
+    (testpath/"hello.d").write <<~EOS
+      import std.stdio;
+      void main()
+      {
+        writeln("Hello world!");
+      }
+    EOS
+    assert_equal "Hello world!", shell_output("#{bin}/rdmd #{testpath}/hello.d").chomp
+
+    # Test strtod/strtof (temporary test)
+    (testpath/"test.c").write <<~EOS
+      #include <stdio.h>
+      #include <stdlib.h>
+      #include <errno.h>
+
+      int main() {
+          float float1 = strtof("0x0.8p-126f", NULL);
+          printf("%s\\n", (float1 != 0) ? "true" : "false");
+          printf("%d\\n", errno);
```

Comment:
Looks like this is throwing `ERANGE 34 Result too large`

Reply:
Nice, that's what I was hoping to catch. Will forward this upstream.